src/Core/Form/IdentifiableObject/DataProvider/ProductFormDataProvider.php

```diff
@@ -304,6 +316,39 @@ private function extractCustomizationsData(ProductForEditing $productForEditing)
         ];
     }
 
+    /**
+     * @param int $productId
+     *
+     * @return array<int, array<string, mixed>>
```

Comment:
```suggestion
     * @return array<string, array<int, array<string, bool|string|int>>
```

Reply:
bool|string|int is kinda equivalent to mixed :smile: 
Altough the one type "array" is misisng, so you think its worth to specify this?